### PR TITLE
Us002b view collection exercises

### PR DESCRIPTION
**Pull request by andrewmil**
Added endpoint to return the data required for the single survey view page in the internal UI.

This includes survey data (short/longName, surveyRef and legalBasis(not yet implemented))
This also includes a list of collection exercises information

[trello](https://trello.com/c/TZoUzqkn/24-us002b-view-collection-exercises-13pts)

### CODE REVIEW

File: ras_backstage/resources/survey/get_survey_by_short_name.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from structlog import wrap_logger
+
+from ras_backstage import survey_api
+from ras_backstage.controllers import collection_exercise_controller, survey_controller
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+@survey_api.route('/shortname/<string:short_name>')
```
Comment (benjefferies) [inline ras_backstage/resources/survey/get_survey_by_short_name.py]: It's worth bearing in mind if we need to get this information by any other query i.e. survey id we could revisit this API and make it based on query params

File: test/resources/test_survey.py [hunk 2]
```diff
@@ -53,3 +69,35 @@ def test_get_survey_list_JSONDecodeError(self, mock_request):
         self.assertEqual(response.status_code, 500)
         self.assertIn('"json": "aaaa"'.encode(), response.data)
         self.assertIn('"message": "Expecting value"'.encode(), response.data)
+
+    @requests_mock.mock()
```
Comment (benjefferies) [inline test/resources/test_survey.py]: Maybe worth adding a test for the empty data scenario

**Comment by andrewmil** [comment]
[US002b: View Collection Exercise(s) (13pts)](https://trello.com/c/TZoUzqkn/24-us002b-view-collection-exercises-13pts)

File: ras_backstage/controllers/collection_exercise_controller.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+from ras_backstage.common.requests_handler import request_handler
+from ras_backstage.exception.exceptions import ApiError
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def get_collection_exercises_by_survey(survey_id):
+    logger.debug('Retrieving collection exercises', survey_id=survey_id)
+    url = f'{app.config["RM_COLLECTION_EXERCISE_SERVICE"]}collectionexercises/survey/{survey_id}'
```
Comment (rhiGriff) [inline ras_backstage/controllers/collection_exercise_controller.py]: Shouldn't we be consistent with these in that we either use .format or this method throughout the whole app?

Comment (jcox-dev) [reply ras_backstage/controllers/collection_exercise_controller.py]: Consistency is preferred but as f-strings are not intended to replace str.format but provide an alternative short-hand we can have a mix (especially for different use-cases).

"[This PEP does not propose to remove or deprecate any of the existing string formatting mechanisms](https://www.python.org/dev/peps/pep-0498/#abstract)."
